src/main/java/com/github/dockerjava/api/model/Frame.java

```diff
@@ -0,0 +1,47 @@
+package com.github.dockerjava.api.model;
+
+import java.util.Arrays;
+
+/**
+ * Represents a logging frame.
+ */
+public class Frame {
+    private final StreamType streamType;
+    private final byte[] payload;
+
+    public Frame(StreamType streamType, byte[] payload) {
+        this.streamType = streamType;
+        this.payload = payload;
+    }
+
+    public StreamType getStreamType() {
+        return streamType;
+    }
+
+    public byte[] getPayload() {
+        return payload;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%s: %s", streamType, new String(payload).trim());
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Frame frame = (Frame) o;
+
+        return Arrays.equals(payload, frame.payload) && streamType == frame.streamType;
```

Comment:
It probably doesn't really matter, but comparing `streamType` could be done first because `Arrays.equals` is much more expensive to do.